### ISSUE TITLE
Implement TensorBoard Logging for Better Experiment Tracking

### ISSUE THREAD

**Issue by Zinnia6277** [kwuking/TimeMixer#139]
**Description:**
Currently, the training progress is only logged to the console, making it difficult to compare different model configurations or track performance across epochs visually. Adding TensorBoard support would significantly improve experiment tracking and visualization capabilities.

**Current Limitations:**
- No visual tracking of metrics during training
- Difficult to compare performance between model configurations
- Cannot easily visualize model architecture and resource usage
- Loss curves and validation metrics are not easily accessible after training

**Proposed Solution:**
1. Integrate TensorBoard logging into the training loop
2. Log key metrics including:
   - Training and validation loss
   - Evaluation metrics (MSE, MAE, etc.)
   - Learning rate changes
   - Model parameter histograms
   - Example predictions vs. ground truth at intervals
3. Add optional model graph visualization

**Implementation Example:**
```python
# In exp/exp_basic.py or specific experiment files
from torch.utils.tensorboard import SummaryWriter

# In __init__ method
self.writer = SummaryWriter(f'runs/{setting}') if use_tensorboard else None

# In training loop
if self.writer:
    self.writer.add_scalar('Loss/train', train_loss, epoch)
    self.writer.add_scalar('Metrics/MSE', mse, epoch)
    self.writer.add_scalar('Metrics/MAE', mae, epoch)
    self.writer.add_scalar('LR', scheduler.get_last_lr()[0], epoch)
    
    # Add histograms periodically
    if epoch % 10 == 0:
        for name, param in self.model.named_parameters():
            self.writer.add_histogram(name, param, epoch)
```

**Benefits:**
- Visual tracking of training progress
- Easy comparison between different model configurations
- Better insights into model convergence patterns
- Improved debugging of training issues
- Enhanced reporting and sharing of experimental results

**Comment by kevinliu2000** [comment]
Thank you for your suggestion. We will consider implementing this feature in the future. You are also welcome to add TensorBoard support yourself. If you have an implementation, please feel free to submit a pull request. We will consider merging it.
